src/store/StudyLegendStore.js

```diff
@@ -338,66 +338,68 @@ export default class StudyLegendStore {
     }
 
     handleDrawPanels = () => {
-        const panelsLen = Object.keys(this.stx.panels).length;
-        Object.keys(this.stx.panels).forEach((id, index) => {
-            if (index === 0) {
-                return;
-            }
-
-            const panelObj = this.stx.panels[id];
-            if (this.mainStore.chart.isMobile) {
-                if (panelObj.up.className.indexOf('show') !== -1) {
-                    panelObj.up.className = 'stx-btn-panel';
-                }
-                if (panelObj.down.className.indexOf('show') !== -1) {
-                    panelObj.down.className = 'stx-btn-panel';
-                }
-                if (panelObj.solo.className.indexOf('show') !== -1) {
-                    panelObj.solo.className = 'stx-btn-panel';
+        if (this.stx) {
+            const panelsLen = Object.keys(this.stx.panels).length;
```

Comment:
why not just store `Object.keys(this.stx.panels)` in the variable to reuse in line 343?